### PR TITLE
[HttpClient] Fix missing metric during network failures

### LINKED ISSUE #4055: [HttpClient] `http.client.duration` metric is not reported in case of network connectivity or dns failure
Metric is reported by extracting HttpResponseMessage from the payload. https://github.com/open-telemetry/opentelemetry-dotnet/blob/00314ac693d6e3efe55a13361a2e5e7e103595da/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerMetricsDiagnosticListener.cs#L49

However, in case of network connectivity failure the response message itself is null and hence no metric will be reported in this case. Alternatively, We can fetch [request](https://github.com/dotnet/runtime/blob/655973db031ed341c8a8e163e59346a8a1c5b367/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L176-L187) from the payload and report the metric.

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #4055 

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.Http.Tests/http-out-test-cases.json [hunk 1]
```diff
@@ -90,35 +90,37 @@
   {
     "name": "Call that cannot resolve DNS will be reported as error span",
     "method": "GET",
-    "url": "https://sdlfaldfjalkdfjlkajdflkajlsdjf.sdlkjafsdjfalfadslkf.com/",
+    "url": "http://sdlfaldfjalkdfjlkajdflkajlsdjf:{port}/",
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.Http.Tests/http-out-test-cases.json:93]: why changed these?

File: src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Fixed an issue of missing `http.client.duration` metric data in case of
+network failures.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md:6]:
```suggestion
network failures (when response is not available).
```

**Review by cijothomas — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.Http.Tests/http-out-test-cases.json (hunk 1 above)
Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.Http.Tests/http-out-test-cases.json:93]: wanted to validate we capture port information as well.

**Pull request merged**